Added planner specific parameters to ompl_planning.yaml emitter.

drchrislewis opened the pull request:
Each parameter is set to current defaults. This is fragile, as defaults may change.
However, yaml makes it difficult to provide correct syntax for parameters as comments which would have been preferable. 

Each parameter also has comments which help users understand what it means.

Commit:
Added planner specific parameters to ompl_planning.yaml emitter.
Each parameter is set to current defaults. This is fragile, as defaults may change.

isucan commented on include/moveit/setup_assistant/tools/moveit_config_data.h:
```diff
@@ -76,6 +76,47 @@ struct GroupMetaData
+/**
+ * Planning parameters which may be set in the config files
+ */
+ struct OmplPlanningParameter{
```
This is not so important, but i like spaces before {

drchrislewis replied:
I inserted either a space or a carriage return before {

FYI I plan to respond individually to each email notice I get. I'm fixing
all your request at once. In the future, I'd be happy to respond to blocks
of request with a single email if you'd like that better.

On Thu, May 15, 2014 at 11:50 AM, Ioan A Sucan notifications@github.comwrote:

> In include/moveit/setup_assistant/tools/moveit_config_data.h:
> 
> > @@ -76,6 +76,47 @@ struct GroupMetaData
> >    int kinematics_solver_attempts_; // solver attempts
> >  };
> > 
> > +/**
> > - \* Planning parameters which may be set in the config files
> > - */
> > - struct OmplPlanningParameter{
> 
> This is not so important, but i like spaces before {
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-planning/moveit_setup_assistant/pull/85/files#r12700022
> .

isucan commented on include/moveit/setup_assistant/tools/moveit_config_data.h:
```diff
@@ -76,6 +76,47 @@ struct GroupMetaData
+ *  @param name: name of planner
+ *  @parameter type: type of planner
+ */
+   OMPLPlannerDescription(const std::string  & name, const std::string  & type){
```
And we use & next to the variable name in MoveIt.

drchrislewis replied:
fixed. It does look better this way.

On Thu, May 15, 2014 at 11:51 AM, Ioan A Sucan notifications@github.comwrote:

> In include/moveit/setup_assistant/tools/moveit_config_data.h:
> 
> > - \* Planning parameters which may be set in the config files
> > - */
> > - struct OmplPlanningParameter{
> > -   std::string name; // name of parameter
> > -   std::string value; // value parameter will receive (but as a string)
> > -   std::string comment; // comment briefly describing what this parameter does
> > - };
> >   +
> >   +/*\* \brief This class describes the OMPL planners by name, type, and parameter list, used to create the ompl_planning.yaml file */
> > - class OMPLPlannerDescription{
> > - public:
> >   +/*\* \brief Constructor
> > - \*  @param name: name of planner
> > - \*  @parameter type: type of planner
> > - */
> > -   OMPLPlannerDescription(const std::string  & name, const std::string  & type){
> 
> And we use & next to the variable name in MoveIt.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-planning/moveit_setup_assistant/pull/85/files#r12700030
> .

isucan commented on src/tools/moveit_config_data.cpp:
```diff
@@ -296,18 +296,87 @@ bool MoveItConfigData::outputOMPLPlanningYAML( const std::string& file_path )
+  std::vector<OMPLPlannerDescription> planner_des;
+
+  OMPLPlannerDescription SBL("SBL","geometric");
+  SBL.addParameter("range","0.0","Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()");
```
Please add a space after every ,

drchrislewis replied:
fixed. I usually do this, not sure why I slipped up here.

On Thu, May 15, 2014 at 11:51 AM, Ioan A Sucan notifications@github.comwrote:

> In src/tools/moveit_config_data.cpp:
> 
> > @@ -296,18 +296,87 @@ bool MoveItConfigData::outputOMPLPlanningYAML( const std::string& file_path )
> > 
> >    emitter << YAML::Value << YAML::BeginMap;
> > -  // Add Planners
> > -  static const std::string planners[] =
> > 
> > ## \-    {"SBL", "EST", "LBKPIECE", "BKPIECE", "KPIECE", "RRT", "RRTConnect", "RRTstar", "TRRT", "PRM", "PRMstar" };
> > -  std::vector<OMPLPlannerDescription> planner_des;
> >   +
> > -  OMPLPlannerDescription SBL("SBL","geometric");
> > -  SBL.addParameter("range","0.0","Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()");
> 
> Please add a space after every ,
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-planning/moveit_setup_assistant/pull/85/files#r12700057
> .

isucan commented:
+1, assuming the build error is fixed.

Commit:
Added space after every , in function calls
Added either a space or a c-return before opening {
Moved & next to the variable in the member function declarations

drchrislewis commented:
please explain both the +1, and the build error.
My intuition says +1 is same as kudos.

It seems to build, run, and the resulting file works with the GUI.
The main issue I have is making sure the parameter names an default values
I provide are identical to those in the source of ompl.
I tried not to change the default behavior. Toward this goal I would have
liked to have generated commented parameters like:

# range: 0.0  #  Maximum distance from current graph to grow ....

However, yaml's emitter did not want to do that... at all.

On Thu, May 15, 2014 at 11:52 AM, Ioan A Sucan notifications@github.comwrote:

> +1, assuming the build error is fixed.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ros-planning/moveit_setup_assistant/pull/85#issuecomment-43235624
> .

isucan commented:
+1 means that i agree this should be merged.
the Travis build that run for your previous version had failed (I did not check why), but it passes now. I will let someone else complete the merge, as this is a config change, and more eyes are better :)

davetcoleman commented:
Looks good
